tests/experimental_combination/attenuator_test.py

```diff
@@ -0,0 +1,127 @@
+from unittest.mock import MagicMock, patch
+
+from wzdx.experimental_combination import attenuator
+from wzdx.tools import polygon_tools
+import json
+
+
+def test_validate_directionality_valid():
+    geotab = {'avl_location': {'position': {'bearing': 0}}}
+    planned_event = {'features': [
+        {'properties': {'core_details': {'direction': 'northbound'}}}]}
+
+    actual = attenuator.validate_directionality(geotab, planned_event)
+
+    assert actual == True
+
+
+def test_validate_directionality_invalid():
+    geotab = {'avl_location': {'position': {'bearing': 180}}}
+    planned_event = {'features': [
+        {'properties': {'core_details': {'direction': 'northbound'}}}]}
+
+    actual = attenuator.validate_directionality(geotab, planned_event)
+
+    assert actual == False
+
+
+def test_get_combined_events_valid():
+    geotab_msgs = json.loads(open('./tests/data/geotab_msgs_1.json').read())
+    wzdx_msgs = [json.loads(open('./tests/data/wzdx_1.json').read())]
+    combined_events = attenuator.get_combined_events(
+        geotab_msgs, wzdx_msgs)
+
+    assert len(combined_events) == 1
+
+
+def test_get_combined_events_valid_multiple():
+    geotab_msgs = json.loads(open('./tests/data/geotab_msgs_1.json').read())
+    wzdx_msgs = [json.loads(open('./tests/data/wzdx_1.json').read())]
+
+    combined_events = attenuator.get_combined_events(
+        geotab_msgs, wzdx_msgs)
+
+    assert len(combined_events) == 1
+
+
+def test_identify_overlapping_features_valid():
+    geotab_msgs = [
+        {
+            "avl_location": {
+                "position": {
+                    "latitude": 39.739928,
+                    "longitude": -104.593591,
+                    "bearing": 45,
+                    "speed": 5,
+                    "odometer": None
+                },
+            },
+            "rtdh_message_id": "110e5ecc-cd7f-48dc-8e59-ac6ef49fc1d0",
+            "rtdh_timestamp": "2022-07-28T21:05:53Z"
+        }]
+    wzdx_msgs = [
+        {
+            'features': [
+                {
+                    'id': 42,
+                    'geometry': {
+                        'coordinates': [
+                            [-104.599491, 39.740070],
+                            [-104.584285, 39.739899]
+                        ]}
+                }
+            ]
+        }
+    ]
+
+    features = attenuator.identify_overlapping_features(geotab_msgs, wzdx_msgs)
+    assert len(features) == 1
+
+
+@patch.object(attenuator, 'get_geometry_for_distance_ahead')
+def test_combine_with_wzdx_ordered(atten_patch):
```

Comment:
Looks like we're only checking a subroutine was called properly, did we want to also check the returned value?

Reply:
Added a returned value check